chocolatey/tools/installer.py

```diff
@@ -0,0 +1,388 @@
+            progress = math.floor(100 * size / total)
+
+        out = sys.stdout
+        if self.started:
+            out.write("\b" * 4)
+
+        out.write("{:3d}%".format(progress))
+        out.flush()
+
+        self.started = True
+
+    def finish(self):
+        sys.stdout.write("\n")
+
+    def __enter__(self):
+        return self.progress
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.finish()
+
+
+class FetchTemp(object):
+    """
+    Fetches the given URL into a temporary file.
+    To be used as a context manager.
+    """
+
+    def __init__(self, url):
+        self.url = url
+
+        fname = os.path.basename(url)
+        root, ext = os.path.splitext(fname)
+        self.filename = tempfile.mktemp(prefix=root + "-", suffix=ext)
+
+    def fetch(self):
+        with self.Progress() as hook:
+            urlretrieve(self.url, self.filename, reporthook=hook)
+
+    def cleanup(self):
+        os.remove(self.filename)
+
+    def __enter__(self):
+        self.fetch()
+
+        return self.filename
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.cleanup()
+
+
+def create_virtualenv(target_dir):
+    """
+    Tries to create a virtualenv by using the built-in `venv` module,
+    or using the `virtualenv` executable if present, or falling back
+    to downloading the official zipapp.
+    """
+
+    def use_venv():
+        try:
+            import venv
+        except ImportError:
+            return
+
+        # on Debian and Ubuntu systems Python is missing `ensurepip`,
+        # prompting the user to install `python3-venv` instead.
+        #
+        # we could handle this, but we'll just let the command fail
+        # and have the users install the package themselves.
+
+        return call([sys.executable, "-m", "venv", target_dir])
+
+    def use_virtualenv():
+        venv_exec = which("virtualenv")
+        if not venv_exec:
+            return
+
+        return call([venv_exec, "-p", sys.executable, target_dir])
+
+    def use_zipapp():
+        print("Downloading virtualenv: ", end="")
+        with FetchTemp(VIRTUALENV_URL) as zipapp:
+            return call([sys.executable, zipapp, target_dir])
+
+    print("Installing virtual environment...")
+    for func in use_venv, use_virtualenv, use_zipapp:
+        retval = func()
+        if retval is None:
+            # command did not run
+            continue
+        if retval == 0:
+            # command successful
+            return
+        # else...
+        sys.exit(1)
+
+
+def get_pip(lib_dir):
+    return (
+        os.path.join(lib_dir, "Scripts", "pip.exe") if IS_WIN
+        else os.path.join(lib_dir, "bin", "pip")
+    )
+
+
+def install_lektor(lib_dir):
+    create_virtualenv(lib_dir)
+
+    pip = get_pip(lib_dir)
+
+    args = [pip, "install"]
+    if IS_WIN:
+        # avoid fail due to PEP 517 on windows
+        args.append("--prefer-binary")
+    args.extend(["--upgrade", "Lektor"])
```

Comment:
We should lock the Lektor version to the package version.

Reply:
Understood. Again, by default chocolatey thinks its version `1.0.0`. I'll have to bumped to `3.2.0`.